Standards/CCSMB-12.md

```diff
@@ -0,0 +1,90 @@
+  "size": [1,1,1],        // (RECOMMENDED) Size in number of blocks required for the build, not needed for printing but is an indicator for an automated builder
+  "isButton": false,      // (OPTIONAL) Global setting for all blocks (see 3dj format)
+  "collideWhenOn": true,  // (OPTIONAL) Global setting for all blocks (see 3dj format)
+  "collideWhenOff": true, // (OPTIONAL) Global setting for all blocks (see 3dj format)
+  "lightLevel": 0,        // (OPTIONAL) Global setting for all blocks (see 3dj format)
+  "redstoneLevel": 0,     // (OPTIONAL) Global setting for all blocks (see 3dj format)
+  "lightWhenOn": false,   // (OPTIONAL) Global setting for all blocks (see 3dj format)
+  "lightWhenOff": false,  // (OPTIONAL) Global setting for all blocks (see 3dj format)
+  "seatPos": undefined,   // (OPTIONAL) Global setting for all blocks (see 3dj format)
+  "presets": {            // (RECOMMENDED) Dictionary of presets, used to define the texture and tint of a shape for compresion and ease of recoloring
+    "presetname": {
+      "texture": "",      // (OPTIONAL) Texture of the preset (see 3dj format)
+      "tint": "FFFFFF"    // (OPTIONAL) Tint of the preset (see 3dj format)
+    },
+  },
+  "blocks": [
+    {
+      "label": "...",                // (OPTIONAL) The name scheme of a block, if empty the "blocks[i].label" SHALL be "Untitled", the string MUST be processed to make use of the Label Variables below
+      "overrideGlobalLabel": false,  // (OPTIONAL) If true, label of this block SHALL override the global label entirely
+      "tooltip": "...",              // (OPTIONAL) The description of the block, overrides the tooltip of the build
+      "coordinates": [[0, 0, 0]],    // Coordinates of the block (if not, automated building programs SHALL fail and a printer SHOULD only print the block once)
+      "amount": 64,                  // SHOULD NOT be used if coordinates is set, but if it is, SHOULD match their amount of sets. Can be used for mass production.
+      "lightLevel": 2,               // (OPTIONAL) The light level of the block, overrides the global setting
+      "redstoneLevel": 14,           // (OPTIONAL) The redstone level of the block, overrides the global setting
+      //Any global setting can be overridden here
+      "shapesOff": [
+        { "bounds": [0, 0, 0, 16, 16, 16], "preset": "presetname"},           //A shape can use a preset, which MUST be defined in presets dictionary
+        { "bounds": [0, 0, 0, 16, 16, 16], "texture": "", "tint": "FFFFFF" }, //Or define its own texture and tint
+      ],
+      "shapesOn": [
+        { /* ... */ }
+      ]
+    }
+  ]
+}
+```
+The preset dictionary and array of coordinates per block's goal is to achieve some sort of compression within the file.
+This format provides enough data for an automated building system.
+
+### Label Variables
+This list shows substitutions that should be made by Readers during printing.
```

Comment:
Be a bit more specific about substitutions. What is being substituted? Should a printer abort if an invalid value is present?

Also, a minor wording change to avoid confusion:

```
This list shows substitutions that Readers SHOULD make by [method] during printing.
```

Reply:
I've clarified the concept in the latest commit